src/main/java/org/broadinstitute/hellbender/Main.java

```diff
@@ -112,29 +136,28 @@ public Object instanceMain(final String[] args) {
      * Note: this is the only method that is allowed to call System.exit (because gatk tools may be run from test harness etc)
      */
     protected final void mainEntry(final String[] args) {
+        final CommandLineProgram program = extractCommandLineProgram(args, getPackageList(), getClassList(), getCommandLineName());
         try {
-            final Object result = instanceMain(args);
+            final Object result = runCommandLineProgram(program, args);
             handleResult(result);
             System.exit(0);
         } catch (final CommandLineException e){
-            //the usage has already been printed so don't print it here.
-            if(printStackTraceOnUserExceptions()) {
-                e.printStackTrace();
+            if (program != null) { //program can be null if th
+                System.err.println(program.getUsage());
```

Comment:
The comment trails off a bit at the end... Having said that I don't see a code path that gets here with program==null, since runCommandLineProgram just returns if it doesn't have a program. I suppose that could change in the future though.